# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/mp4parse/src/tests.rs
# Commit: 9493029c76b9
# Full Hash: 9493029c76b93861ca10dbfb29d4178773144416
# Author: zaggy1024 <Zaggy1024@gmail.com>
# Date: 2023-01-28 21:11:06
# Regressor Bug: 1788119
# File Overlap Count: 1
# Description:
#   Bug 1788119 - Part 4 - Update mp4parse-rust for AVIS support. r=kinetik,glandium,supply-chain-reviewers
#   
#   Depends on D156652
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D156653
# ==============================================================================

diff -r 7d5cc211e3f5 -r 9493029c76b9 third_party/rust/mp4parse/src/tests.rs
--- a/third_party/rust/mp4parse/src/tests.rs	Fri Jan 27 19:54:29 2023 +0000
+++ b/third_party/rust/mp4parse/src/tests.rs	Fri Jan 27 19:54:29 2023 +0000
@@ -6,17 +6,16 @@
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
 use super::read_mp4;
-use super::Error;
 use super::ParseStrictness;
+use super::{Error, Status};
 use fallible_collections::TryRead as _;
 
 use std::convert::TryInto as _;
 use std::io::Cursor;
 use std::io::Read as _;
-extern crate test_assembler;
-use self::test_assembler::*;
+use test_assembler::*;
 
-use boxes::BoxType;
+use crate::boxes::BoxType;
 
 enum BoxSize {
     Short(u32),
@@ -122,7 +121,7 @@
 fn read_box_header_short_invalid_size() {
     let mut stream = make_box(BoxSize::UncheckedShort(2), b"test", |s| s);
     match super::read_box_header(&mut stream) {
-        Err(Error::InvalidData(s)) => assert_eq!(s, "malformed size"),
+        Err(Error::InvalidData(s)) => assert_eq!(s, Status::BoxBadSize),
         _ => panic!("unexpected result reading box with invalid size"),
     };
 }
@@ -131,7 +130,7 @@
 fn read_box_header_long_invalid_size() {
     let mut stream = make_box(BoxSize::UncheckedLong(2), b"test", |s| s);
     match super::read_box_header(&mut stream) {
-        Err(Error::InvalidData(s)) => assert_eq!(s, "malformed wide size"),
+        Err(Error::InvalidData(s)) => assert_eq!(s, Status::BoxBadWideSize),
         _ => panic!("unexpected result reading box with invalid size"),
     };
 }
@@ -490,7 +489,10 @@
     let mut stream = iter.next_box().unwrap().unwrap();
     assert_eq!(stream.head.name, BoxType::HandlerBox);
     assert_eq!(stream.head.size, 45);
-    assert!(super::read_hdlr(&mut stream, ParseStrictness::Strict).is_err());
+    assert_eq!(
+        super::Status::from(super::read_hdlr(&mut stream, ParseStrictness::Strict)),
+        super::Status::HdlrNameMultipleNul,
+    );
 }
 
 #[test]
@@ -515,13 +517,10 @@
     let mut stream = iter.next_box().unwrap().unwrap();
     assert_eq!(stream.head.name, BoxType::HandlerBox);
     assert_eq!(stream.head.size, 32);
-    match super::read_hdlr(&mut stream, ParseStrictness::Normal) {
-        Err(Error::Unsupported(msg)) => assert_eq!("hdlr version", msg),
-        result => {
-            eprintln!("{:?}", result);
-            panic!("expected Error::Unsupported")
-        }
-    }
+    assert_eq!(
+        super::Status::from(super::read_hdlr(&mut stream, ParseStrictness::Normal)),
+        super::Status::HdlrUnsupportedVersion,
+    );
 }
 
 #[test]
@@ -533,17 +532,10 @@
     let mut stream = iter.next_box().unwrap().unwrap();
     assert_eq!(stream.head.name, BoxType::HandlerBox);
     assert_eq!(stream.head.size, 32);
-    match super::read_hdlr(&mut stream, ParseStrictness::Strict) {
-        Err(Error::InvalidData(msg)) => assert_eq!(
-            "The HandlerBox 'pre_defined' field shall be 0 \
-             per ISOBMFF (ISO 14496-12:2020) ยง 8.4.3.2",
-            msg
-        ),
-        result => {
-            eprintln!("{:?}", result);
-            panic!("expected Error::InvalidData")
-        }
-    }
+    assert_eq!(
+        super::Status::from(super::read_hdlr(&mut stream, ParseStrictness::Strict)),
+        super::Status::HdlrPredefinedNonzero,
+    );
 }
 
 #[test]
@@ -555,17 +547,10 @@
     let mut stream = iter.next_box().unwrap().unwrap();
     assert_eq!(stream.head.name, BoxType::HandlerBox);
     assert_eq!(stream.head.size, 32);
-    match super::read_hdlr(&mut stream, ParseStrictness::Strict) {
-        Err(Error::InvalidData(msg)) => assert_eq!(
-            "The HandlerBox 'reserved' fields shall be 0 \
-             per ISOBMFF (ISO 14496-12:2020) ยง 8.4.3.2",
-            msg
-        ),
-        result => {
-            eprintln!("{:?}", result);
-            panic!("expected Error::InvalidData")
-        }
-    }
+    assert_eq!(
+        super::Status::from(super::read_hdlr(&mut stream, ParseStrictness::Strict)),
+        super::Status::HdlrReservedNonzero,
+    );
 }
 
 #[test]
@@ -577,17 +562,10 @@
     let mut stream = iter.next_box().unwrap().unwrap();
     assert_eq!(stream.head.name, BoxType::HandlerBox);
     assert_eq!(stream.head.size, 32);
-    match super::read_hdlr(&mut stream, ParseStrictness::Normal) {
-        Err(Error::InvalidData(msg)) => assert_eq!(
-            "The HandlerBox 'name' field shall be null-terminated \
-             per ISOBMFF (ISO 14496-12:2020) ยง 8.4.3.2",
-            msg
-        ),
-        result => {
-            eprintln!("{:?}", result);
-            panic!("expected Error::InvalidData")
-        }
-    }
+    assert_eq!(
+        super::Status::from(super::read_hdlr(&mut stream, ParseStrictness::Normal)),
+        super::Status::HdlrNameNoNul,
+    );
 }
 
 #[test]
@@ -1278,7 +1256,7 @@
     let mut stream = iter.next_box().unwrap().unwrap();
 
     match super::read_esds(&mut stream) {
-        Err(Error::InvalidData(s)) => assert_eq!(s, "Invalid descriptor."),
+        Err(Error::InvalidData(s)) => assert_eq!(s, Status::EsdsBadDescriptor),
         _ => panic!("unexpected result with invalid descriptor"),
     }
 }